kubernetes/resource_kubernetes_labels.go

```diff
@@ -190,6 +221,42 @@ func getManagedLabels(managedFields []v1.ManagedFieldsEntry, manager string) (ma
 	return labels, nil
 }
 
+// getTemplateManagedLabels reads the field manager metadata to discover which fields we're managing
+func getTemplateManagedLabels(managedFields []v1.ManagedFieldsEntry, manager string, kind string) (map[string]interface{}, error) {
+	var labels map[string]interface{}
+	for _, m := range managedFields {
+		if m.Manager != manager {
+			continue
+		}
+		var mm map[string]interface{}
+		err := json.Unmarshal(m.FieldsV1.Raw, &mm)
+		if err != nil {
+			return nil, err
+		}
+		var spec map[string]interface{}
+		if s, ok := mm["f:spec"].(map[string]interface{}); ok {
+			spec = s
+		}
+		if kind == "CronJob" {
+			if jt, ok := spec["f:jobTemplate"].(map[string]interface{}); ok {
+				spec = jt["f:spec"].(map[string]interface{})
+			}
+		}
+		var template map[string]interface{}
+		if t, ok := spec["f:template"].(map[string]interface{}); ok {
+			template = t
+		}
+		var metadata map[string]interface{}
+		if mmm, ok := template["f:metadata"].(map[string]interface{}); ok {
+			metadata = mmm
+		}
+		if l, ok := metadata["f:labels"].(map[string]interface{}); ok {
+			labels = l
```

Comment:
> Since `labels` is declared outside the for loop, this will overwrite it every time there is a match on `manager`. Should this rather be an append-type operation, adding the contents of `l` to `labels`?

I have directly copied from template_annotations for this part, I will check for this.

Reply:
Hi @alexsomesan I’ve done some investigating and I think there could be potential issues if the Kubernetes API returns a configuration like the one below:

```
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: test-cm
  namespace: default
  labels:
    test-label: test
  managedFields:
  - manager: kubectl
    operation: Apply # note capitalization: "Apply" (or "Update")
    apiVersion: v1
    time: "2010-10-10T0:00:00Z"
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:labels:
          f:test-label: {}
      f:data:
        f:key: {}
  - manager: kubectl
    operation: Update # note capitalization: "Apply" (or "Update")
    apiVersion: v1
    time: "2010-10-10T0:00:00Z"
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:labels:
          f:test-label: {}
      f:data:
        f:key: {}
data:
  key: some value
```
However, I'm finding it difficult to reproduce this configuration. When I attempt to update a ConfigMap using kubectl apply and then reapply, it appears that the API consolidates the two operations under one manager. Do you have any suggestions on how to reproduce this issue?

Reply:
> Since `labels` is declared outside the for loop, this will overwrite it every time there is a match on `manager`. Should this rather be an append-type operation, adding the contents of `l` to `labels`?

Hi @alexsomesan yeah, I agree there could be potential issues in theory, however, I wasn't able to reproduce it with `kubectl`. 

I tried `kubectl apply` and `kubectl patch` on same object multiple times, however, looks like the server will store different manager `kubectl-client-side-apply` and `kubectl-patch`.

Do you have any suggestions on reproducing this issue? Or you guys are fine for me to change the code (without testing this case).

Reply:
Hi @theloneexplorerquest, this issue is specific to using `server-side-apply` to access the API, however it appears you may be using `client-side-apply` as per the name of manager. `kubectl` by default uses `client-side-apply` and requires you to explicitly state `--server-side` in order to use the `server-side-apply`.

Reply:
> Hi @theloneexplorerquest, this issue is specific to using `server-side-apply` to access the API, however it appears you may be using `client-side-apply` as per the name of manager. `kubectl` by default uses `client-side-apply` and requires you to explicitly state `--server-side` in order to use the `server-side-apply`. Here's a useful [link](https://kubernetes.io/docs/reference/using-api/server-side-apply/#field-management:~:text=kubectl%20apply%20implementation.-,Field%20management,-The%20Kubernetes%20API).

Hi @sheneska thanks for reply. Yeah, I actually tried `--server-side`, however, I wasn't able to generate manifest with two field managers both named `kubectl`, do you have suggestion for that?